src/org/opensha/commons/calc/magScalingRelations/magScalingRelImpl/TMG2017_MagAreaRel.java

```diff
@@ -0,0 +1,304 @@
+/*******************************************************************************
+ * Copyright 2020 Statement goes here
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package org.opensha.commons.calc.magScalingRelations.magScalingRelImpl;
+
+
+import org.opensha.commons.calc.magScalingRelations.MagAreaRelationship;
+import org.opensha.commons.util.FaultUtils;
+
+
+/**
+ * <b>Title:</b>TMG2017_MagAreaRel<br>
+ *
+ * <b>Description:</b>  
+ * <p> 
+ * This implements the set of magnitude versus rupture area relation of 
+ * Thingbaijam K.K.S., P.M. Mai and K. Goda (2017, Bull. Seism. Soc. Am., 107, 2225–2246. 
+ * 
+ * Addition to rake, we also use regime (="crustal" or "interface") as an additional parameter 
+ * to distinguish shallow crustal and subduction-interface. The default is "crustal". 
+ * However, the relations for normal faults are also applicable to inslab events.   
+ *  
+ * Notes:
+ * (1) Valid rake is in the range -180 to 180 degrees.  
+ * (2) the standard deviation for area as a function of mag is given for log(area) (base-10) not area.
+ * (2) The current implementation does not handle fixed seismogenic width (or saturated rupture width);
+ *     It will require implementing MagLenRel and MagWidRel as well.   
+ *     
+ * Also see: https://github.com/thingbaijam/sceqsrc
+ *    
+ * </p>
+ *
+ * @version 0.0
+ */
+
+public class TMG2017_MagAreaRel extends MagAreaRelationship {
+    /**
+	 * To Do: Write something here.
+	 */
+	private static final long serialVersionUID = 1L;
+	final static String C = "TMG2017_MagAreaRel";
+    public final static String NAME = "Thingbaijam et al.(2017)";
+    protected String regime = "crustal"; 
```

Comment:
`regime` is a prime candidate for using an enumeration. This restricts the possible values automatically, and no change to FaultUtils is necessary.

Example:

```
public enum Regime{
    CRUSTAL,
    INTERFACE
}

protected Regime regime;

public void setRegime(Regime regime){
    this.regime = regime;
}
```

This can then be used as `setRegime(Regime.CRUSTAL);` (uppercase enum values are standard).

Reply:
- [ ]  introduce Regime as enumeration

Reply:
Sure, right!  My concern: wouldn't that be complicated from user of view. User would just like specify which faulting regime: either crustal or interface (perhaps, I should change this to "subduction") in order to apply appropriate scaling relationship.

Reply:
This should be easier from a user's point of view because they can immediately see which values are legal. If it's just a String, then they need to dive into the code or read the documentation.